src/translation/Translation.ts

```diff
@@ -0,0 +1,38 @@
+import trJson from './tr.json'
+
+export class Translation {
+    data: any
+
+    constructor(data: any) {
+        this.data = data
+    }
+
+    tr(key: string, parameters?: string[]): string {
+        var str: string = this.data[key]
+        if (!str) return key
+        if (parameters)
+            for (let i = 0; i < parameters.length; i++) {
+                str = str.replace('%' + (i + 1) + '$s', parameters[i])
+            }
+        return str || ''
+    }
+}
+
+let translation: Translation
```

Comment:
singleton similar to how the stores are handled

Reply:
The other stores are not real singletons though. I didn't make the singletons because they are a little easier to unit test if dependency injection still works